src/GMScriptCallback.cpp

```diff
@@ -64,3 +64,36 @@ static int getGMSelection(lua_State* L)
 /// getGMSelection()
 /// Returns an list of objects that the GM currently has selected.
 REGISTER_SCRIPT_FUNCTION(getGMSelection);
+
+static int onGMClick(lua_State* L)
+{
+    ScriptSimpleCallback callback;
+
+    int idx = 1;
+    convert<ScriptSimpleCallback>::param(L,idx,callback);
+
+    gameGlobalInfo->on_gm_click=callback;
+
+    foreach (Updatable, u, updatableList)
+    {
+        P<GameMasterScreen> game_master_screen = u;
```

Comment:
Instead of this piece of slightly complex and convoluted code, it would be much clearer to hide/show the buttons from the gmscreen update function. That is the way most things work in EE (state based instead of event based)